io/disk/iozone.py

```diff
@@ -0,0 +1,577 @@
+        :param results: List of n lists containing data from performance runs.
+        :param size: Numerical value of a size (say, file_size) that was used
+                     to filter the original results list.
+        :return: List with 1 list containing average data from the performance
+                 run.
+        """
+        average_line = []
+        if size is not None:
+            average_line.append(size)
+        for i in range(2, 15):
+            average = data_structures.geometric_mean(
+                [line[i] for line in results]) / 1024.0
+            average = int(average)
+            average_line.append(average)
+        return average_line
+
+    def process_results(self, results, label=None):
+        """
+        Process a list of IOzone results according to label.
+
+        :label: IOzone column label that we'll use to filter and compute
+                geometric mean results, in practical term either 'file_size'
+                or 'record_size'.
+        :result: A list of n x m columns with original iozone results.
+        :return: A list of n-? x (m-1) columns with geometric averages for
+                values of each label (ex, average for all file_sizes).
+        """
+        performance = []
+        if label is not None:
+            index = _LABELS.index(label)
+            sizes = data_structures.ordered_list_unique(
+                [line[index] for line in results])
+            for size in sizes:
+                r_results = [line for line in results if line[index] == size]
+                performance.append(self.average_performance(r_results, size))
+        else:
+            performance.append(self.average_performance(results))
+
+        return performance
+
+    @staticmethod
+    def parse_file(p_file):
+        """
+        Parse an IOzone results file.
+
+        :param file: File object that will be parsed.
+        :return: Matrix containing IOzone results extracted from the file.
+        """
+        lines = []
+        for line in p_file.readlines():
+            fields = line.split()
+            if len(fields) != 15:
+                continue
+            try:
+                lines.append([int(i) for i in fields])
+            except ValueError:
+                continue
+        return lines
+
+    def report(self, overall_results, record_size_results, file_size_results):
+        """
+        Generates analysis data for IOZone run.
+
+        Generates a report to both logs (where it goes with nice headers) and
+        output files for further processing (graph generation).
+
+        :param overall_results: 1x15 Matrix containing IOzone results for all
+                file sizes
+        :param record_size_results: nx15 Matrix containing IOzone results for
+                each record size tested.
+        :param file_size_results: nx15 Matrix containing file size results
+                for each file size tested.
+        """
+        logger = logging.getLogger()
```

Comment:
The same here. Ideally you should pass the `self.log` from test and use it here as `logger`, or at least please use `getLogger("avocado.test")`. The difference is in the `debug.log` vs. `job.log`.